R/vertica.udx.R

```diff
@@ -0,0 +1,104 @@
+
+
+generate_sql_query <- function(fun, tbl, col_types, col_names, lengths, partition_cols, order_cols, libraries = NULL, ...)
+{
+	
+	params = list(...)
+	params$vertica.dplyr.libraries = libraries
+	params$vertica.dplyr.func = fun
+
+	params = serialize_object(params)
+
+	outcols = list(lengths = lengths, col_types = col_types, names = col_names)
+	outcols = serialize_object(outcols)
+
+	parameters_clause = paste0("using parameters params = '",
+		params, "', outcols='",outcols,"'")
+
+	over_clause = ""
+	
+	source_query <- tbl$query$sql
+	source_query <- paste0("( ", source_query ,") AS ZZZ1")
+
+	query = paste0("select vertica_function_executor(* ", 
+	        parameters_clause,
+		") over(",
+		over_clause,
+		") from ",
+		source_query)
+
+
+	vertica <- tbl$src$con
+	temp <- select(tbl)
+	output_table <- temp
+	output_table$select <- NULL
+	output_table <- update(output_table, select = col_names)
+	output_table$query <- query.VerticaConnection(vertica, query,NULL)
+	output_table$from <- paste0("( ", query ,") AS ZZZ1")
+	class(output_table$from) <- c("sql", "character")
+
+	output_table$select = sapply(col_names,as.symbol)
+	output_table$mutate = TRUE
+	output_table$order_by = FALSE
+	output_table$summarise = FALSE
+	output_table$where = FALSE
+	#print(sqlQuery(vertica@conn,query))
+	
+	return (output_table)
+
+}
+
+
+execute_custom_R_udf <- function( fun, tbl, ..., 
+		     partition_cols = character(0), 
+		     order_cols = character(0), 
+		     libraries = character(0))
+{
+	temp.env <- new.env(parent = globalenv())
+	environment(fun) <- temp.env
+
+	additional_args = list(...)
+	if(any(names(additional_args) == ""))
+		stop("All additional arguments must be named")
+
+	test_fun <- function(x)
```

Comment:
Instead of creating a test_fun to redundantly run the function, I wonder if it's better to just have the user provide the output column names and types manually.